### ISSUE TITLE
Default lsp config overridden when there is a local pyrightconfig.json

### ISSUE THREAD

**Issue by rbhanot4739** [DetachHead/basedpyright#685]
I am using Basedpyright with Neovim and observed an unexpected behaviour where the presence of `pyrightconfig.json` seems to override the default lsp config. Here is my lsp config
```
     {
          root_dir = function(fname)
            return util.root_pattern(unpack(root_files))(fname)
          end,
          settings = {
            basedpyright = {
              analysis = {
                typeCheckingMode = "standard",
                diagnosticSeverityOverrides = {
                  reportUndefinedVariable = false,
                },
              },
            },
          },
        }
```

Since I use ruff for linting, I have disabled the same from pyright config. This works fine when there is no `pyrightconfig.json` in the project root. Consider the following code in a file named `foo.py` .
```
print(token)
```
And here is the diagnostics I get for the above fille. 
```
foo.py  1 
 Undefined name `token` Ruff (F821) [2, 7]
```
So basedpyright didn't reported this error as it was disabled in the lsp config. Now if I create the `pyrightconfig.json` with following contents 
```
{
  "reportMissingTypeStubs": false,
  "typeCheckingMode": "basic"
}
```
this is what happens
```
     foo.py  2 
       "token" is not defined basedpyright (reportUndefinedVariable) [2, 7]
        Undefined name `token` Ruff (F821) [2, 7]
```

I didn't overwrite the key `reportUndefinedVariable` so it should have fallen back to the default lsp config, but it seems config is not getting merged but rather overridden. 
is this a bug or am i doing something wrong here ?
If that's the case then mere presence of a `pyrightconfig.json`  with as minimal config keys will replace all default settings.

**Comment by DetachHead** [comment]
this behaviour comes from upstream and is intentional to prevent project-specific settings from being overridden by user settings.  see the discussion in #513

**Comment by DetachHead** [comment]
merging this issue with #513
